clang/test/SemaSYCL/unsupported_math.cpp

```diff
@@ -0,0 +1,26 @@
+// RUN: %clangxx --sycl %s -c -o a_kernel.bc -Xclang -verify
+
+#include <CL/sycl.hpp>
+
+int run_test_a(int v, cl::sycl::queue &deviceQueue) {
+    try {
+      int arr[] = {v};
+      {
+        cl::sycl::buffer<int, 1> buf(arr, 1);
+        deviceQueue.submit([&](cl::sycl::handler &cgh) {
+          cl::sycl::stream stream(1024, 80, cgh);
+          auto acc = buf.get_access<cl::sycl::access::mode::read_write>(cgh);
+          cgh.single_task<class kernel_a>([=]() {
+            acc[0] *= 2;
+            acc[0] += (int)__builtin_fabsf(-1.0f); // expected-error{{builtin is not supported on this target}}
```

Comment:
Technically speaking, this doesn't check all of the unsupported functions, so there's no "full" regression test. It may be sufficient to check error triggering only, but someone else had better confirm (@bader, @romanovvlad?)

Reply:
This change will most probably be replaced by a more complete solution that will emit this diagnostic for ALL unsupported functions (and not just math functions). We can add a full regression test at that point.

Reply:
Then I'd suggest adding another 1-2 checks for now.
Merely curious, how is the test going to be structured for all functions?

Reply:
I am thinking we will mostly use function pointers.

Reply:
I added couple more checks. Thanks

Reply:
Thanks!